core/src/main/java/org/arquillian/smart/testing/impl/TestStrategyApplierImpl.java

```diff
@@ -57,18 +57,21 @@
     }
 
     private Set<TestSelection> selectTests(Configuration configuration) {
-
-        final List<String> strategies = Arrays.asList(configuration.getStrategies());
-        if (strategies.isEmpty()) {
+        if (configuration.getStrategies().length == 0) {
             logger.warn(
                 "Smart Testing Extension is installed but no strategies are provided. It won't influence the way how your tests are executed. "
                     + "For details on how to configure it head over to http://bit.ly/st-config");
             return Collections.emptySet();
         }
 
+        List<String> errorMessages = new ArrayList<>();
+        configuration.autocorrectStrategies(testExecutionPlannerLoader.getAvailableStrategyNames(), errorMessages);
+        errorMessages.forEach(msg -> logger.error(msg));
+
```

Comment:
We are throwing exception for [DependencyResolver](https://github.com/arquillian/smart-testing/pull/265/files#diff-e5427bc921ef9faf2177273978bd12feR41) if we have any errorMessages.

Is there any reason why are we silently ignoring it in ST api?

Reply:
Well, the usage of the autocorrection in the ST is kind of safety.
There are two usages:
* Maven integration where the first check is performed on the Maven extension side
* Che and other integrations, where the Configuration is loaded before the ST API is called, so the integration can autocorrect the mistyped strategies and throw an exception if needed. 

This line in `TestStrategyApplierImpl.java` class is just for sure that no understandable exception is thrown and just prints the warning. 
This is my understanding and expectations. If there are cases/integrations when the exception should be thrown, then it should do on its own.